**Order and Product analysis using Pandas**

In [121]:
import pandas as pd
#read data from the excel files
data = pd.read_excel('Order_Details.xlsx')
data2 = pd.read_excel('Products.xlsx')
#cheak the columns andalso weather the data contain any null values
print(data.info()) 
print(data2.info())


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2086 entries, 0 to 2085
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   OrderID    2086 non-null   int64  
 1   ProductID  2086 non-null   int64  
 2   UnitPrice  2086 non-null   float64
 3   Quantity   2086 non-null   int64  
 4   Discount   2086 non-null   float64
dtypes: float64(2), int64(3)
memory usage: 81.6 KB
None
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   ProductID        77 non-null     int64  
 1   ProductName      77 non-null     object 
 2   SupplierID       77 non-null     int64  
 3   CategoryID       77 non-null     int64  
 4   QuantityPerUnit  77 non-null     object 
 5   UnitPrice        77 non-null     float64
 6   UnitsInStock     77 non-null     int64  
 7   UnitsOnOrder     77 non-null     int

In [68]:
# merging the two tables using joins
final_table=pd.merge(data,data2,on='ProductID',how='inner')
print(final_table.info())


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2086 entries, 0 to 2085
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   OrderID          2086 non-null   int64  
 1   ProductID        2086 non-null   int64  
 2   UnitPrice_x      2086 non-null   float64
 3   Quantity         2086 non-null   int64  
 4   Discount         2086 non-null   float64
 5   ProductName      2086 non-null   object 
 6   SupplierID       2086 non-null   int64  
 7   CategoryID       2086 non-null   int64  
 8   QuantityPerUnit  2086 non-null   object 
 9   UnitPrice_y      2086 non-null   float64
 10  UnitsInStock     2086 non-null   int64  
 11  UnitsOnOrder     2086 non-null   int64  
 12  ReorderLevel     2086 non-null   int64  
 13  Discontinued     2086 non-null   bool   
dtypes: bool(1), float64(3), int64(8), object(2)
memory usage: 214.0+ KB
None


Here we only choose the coloumns we want 
and the final table contain
    1.OrderID
    2.ProductID
    3.ProductName
    4.UnitPrice
    5.Quantity
    6.Discount
    7.Total_amount

In [69]:

df=pd.DataFrame(final_table)
print(df.head())

   OrderID  ProductID  UnitPrice_x  Quantity  Discount ProductName  \
0    10285          1         14.4        45      0.20        Chai   
1    10294          1         14.4        18      0.00        Chai   
2    10317          1         14.4        20      0.00        Chai   
3    10348          1         14.4        15      0.15        Chai   
4    10354          1         14.4        12      0.00        Chai   

   SupplierID  CategoryID     QuantityPerUnit  UnitPrice_y  UnitsInStock  \
0           1           1  10 boxes x 20 bags         18.0            39   
1           1           1  10 boxes x 20 bags         18.0            39   
2           1           1  10 boxes x 20 bags         18.0            39   
3           1           1  10 boxes x 20 bags         18.0            39   
4           1           1  10 boxes x 20 bags         18.0            39   

   UnitsOnOrder  ReorderLevel  Discontinued  
0             0            10         False  
1             0            10 

In [70]:
df['Total_amount'] = df['UnitPrice_x']*df['Quantity'] - ( df['UnitPrice_x']*df['Quantity']*df['Discount'])


In [71]:
print(df.head())


   OrderID  ProductID  UnitPrice_x  Quantity  Discount ProductName  \
0    10285          1         14.4        45      0.20        Chai   
1    10294          1         14.4        18      0.00        Chai   
2    10317          1         14.4        20      0.00        Chai   
3    10348          1         14.4        15      0.15        Chai   
4    10354          1         14.4        12      0.00        Chai   

   SupplierID  CategoryID     QuantityPerUnit  UnitPrice_y  UnitsInStock  \
0           1           1  10 boxes x 20 bags         18.0            39   
1           1           1  10 boxes x 20 bags         18.0            39   
2           1           1  10 boxes x 20 bags         18.0            39   
3           1           1  10 boxes x 20 bags         18.0            39   
4           1           1  10 boxes x 20 bags         18.0            39   

   UnitsOnOrder  ReorderLevel  Discontinued  Total_amount  
0             0            10         False    518.399998  
1 

In [72]:
df = df[['OrderID','ProductID','ProductName','UnitPrice_x','Quantity','Discount','Total_amount']]

In [73]:
print(df.info())
print(df.head())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2086 entries, 0 to 2085
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   OrderID       2086 non-null   int64  
 1   ProductID     2086 non-null   int64  
 2   ProductName   2086 non-null   object 
 3   UnitPrice_x   2086 non-null   float64
 4   Quantity      2086 non-null   int64  
 5   Discount      2086 non-null   float64
 6   Total_amount  2086 non-null   float64
dtypes: float64(3), int64(3), object(1)
memory usage: 114.2+ KB
None
   OrderID  ProductID ProductName  UnitPrice_x  Quantity  Discount  \
0    10285          1        Chai         14.4        45      0.20   
1    10294          1        Chai         14.4        18      0.00   
2    10317          1        Chai         14.4        20      0.00   
3    10348          1        Chai         14.4        15      0.15   
4    10354          1        Chai         14.4        12      0.00   

   Total_amount  

Average price of the products

In [78]:
avg_unitprice=df['UnitPrice_x'].mean()
print('avg_price = ',avg_unitprice)

avg_price =  25.964098753595398


Total amount generated from these orders 

In [80]:
total = df['Total_amount'].sum()
print('total =',total)

total = 1222636.9233361713


In [85]:
total_orders=df['OrderID'].count()
print('total Orders =',total_orders)

total Orders = 2086


In [96]:
group_products = df.groupby('ProductName')['ProductID'].count()
print(group_products.head(5))

ProductName
Alice Mutton         37
Aniseed Syrup        12
Boston Crab Meat     41
Camembert Pierrot    51
Carnarvon Tigers      9
Name: ProductID, dtype: int64


In [94]:
sorted_sales = group_products.sort_values(ascending=False)
print(sorted_sales.head(10))

ProductName
Raclette Courdavault               54
Gorgonzola Telino                  51
Camembert Pierrot                  51
Guaraná Fantástica                 51
Gnocchi di nonna Alice             50
Tarte au sucre                     48
Jack's New England Clam Chowder    47
Rhönbräu Klosterbier               46
Chang                              44
Pavlova                            43
Name: ProductID, dtype: int64


In [95]:
sorted_sales = group_products.sort_values(ascending=True)
print(sorted_sales.head(10))

ProductName
Mishi Kobe Niku               5
Chocolade                     6
Gravad lax                    6
Genen Shouyu                  6
Louisiana Hot Spiced Okra     8
Schoggi Schokolade            9
Carnarvon Tigers              9
Chef Anton's Gumbo Mix       10
Laughing Lumberjack Lager    10
Valkoinen suklaa             10
Name: ProductID, dtype: int64


In [100]:
min_discount =df['Discount'].min()
max_discount=df['Discount'].max()
print("minimum discount is =",min_discount)
print("maximum discount is =",max_discount)



minimum discount is = 0.0
maximum discount is = 0.25


In [118]:
exp_product=df.loc[df['UnitPrice_x'].idxmax()]
inexp_product=df.loc[df['UnitPrice_x'].idxmin()]
print(f'expencive product is =',exp_product['ProductName'],exp_product['UnitPrice_x'],'$')
print(f'inexpencive product is =',inexp_product['ProductName'],inexp_product['UnitPrice_x'],'$')




expencive product is = Côte de Blaye 263.5 $
inexpencive product is = Geitost 2.0 $


In [119]:
expencive_order = df.loc[df['Total_amount'].idxmax()]
print('the expencive order is =',expencive_order)

the expencive order is = OrderID                 10981
ProductID                  38
ProductName     Côte de Blaye
UnitPrice_x             263.5
Quantity                   60
Discount                  0.0
Total_amount          15810.0
Name: 919, dtype: object


In [120]:
inexpencive_order = df.loc[df['Total_amount'].idxmin()]
print('the inexpencive order is =',inexpencive_order)

the inexpencive order is = OrderID         10462
ProductID          13
ProductName     Konbu
UnitPrice_x       4.8
Quantity            1
Discount          0.0
Total_amount      4.8
Name: 242, dtype: object


**Summary:**

The analysis identified **'Côte de Blaye'** as the most expensive product, priced at $263.50 per unit.

The highest-value order consisted of 60 units of -**'Côte de Blaye,'** resulting in a total order value of $15,810.00.